chore(docs): Old occurrences of gatbyjs.org

AnilSeervi opened the pull request:
<!--
  Have any questions? Check out the contributing docs at https://gatsby.dev/contribute, or
  ask in this Pull Request and a Gatsby maintainer will be happy to help :)
-->

<!--
  Is this a blog post? Check out the docs at https://www.gatsbyjs.com/contributing/blog-contributions/, and please mention if the blog post is pre-approved
  by someone from Gatsby.
-->

## Description
- Fix instances of `gatsby.org`
- Typo: gatbyjs.org -> gatsbyjs.com
- fix 404 links
<!-- Write a brief description of the changes introduced by this PR -->

### Documentation

<!--
  Where is this feature or API documented?

  - If docs exist:
    - Update any references, if relevant. This includes Guides and Gatsby Internals docs.
  - If no docs exist:
    - Create a stub for documentation including bullet points for how to use the feature, code snippets (including from happy path tests), etc.
  - Tag @gatsbyjs/documentation for review, pairing, polishing of the documentation
-->

## Related Issues

<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234

  Link to an issue that is partially addressed by this PR (if there are any)
  e.g. Addresses #1234

  Link to related issues (if there are any)
  e.g. Related to #1234
-->

tyhopp commented:
Hi @AnilSeervi, thanks for the PR!

It looks like the CI didn't run on your PR, could you push an empty commit or [enable us to make edits](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so that we can have it run?

I also had a look at the docs subdir and noticed there were other occurrences of `gatsbyjs.org` (such as [this occurrence](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/gatsby-style-guide.md?plain=1#L45)) that we can change (ignoring store and example urls like using-gatsby-image). If you're interested, it would be great if you could help also change those.

AnilSeervi commented:
> Hi @AnilSeervi, thanks for the PR!
> 
> It looks like the CI didn't run on your PR, could you push an empty commit or [enable us to make edits](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) so that we can have it run?
> 
> I also had a look at the docs subdir and noticed there were other occurrences of `gatsbyjs.org` (such as [this occurrence](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/gatsby-style-guide.md?plain=1#L45)) that we can change (ignoring store and example urls like using-gatsby-image). If you're interested, it would be great if you could help also change those.

I went ahead and fixed all occurences of `gatbsy.org` in _/docs_

Also fixed some 404 links while I was at it ✨

tyhopp reviewed (approved):
Looks great, thanks @AnilSeervi!

tyhopp commented on docs/docs/how-to/styling/using-web-fonts.md:
```diff
@@ -18,7 +18,7 @@ Some examples of web font services include [Google Fonts](https://fonts.google.c
 ### How to use Web Font Loader with Typekit
 
-You can add Typekit Web Fonts to your project by using the [gatsby-plugin-web-font-loader](https://www.gatsbyjs.org/packages/gatsby-plugin-web-font-loader/?=font) and your [Adobe Fonts project id](https://fonts.adobe.com/my_fonts#web_projects-section). For example, this is how you can add Futura to your project.
+You can add Typekit Web Fonts to your project by using the [gatsby-plugin-web-font-loader](https://www.gatsbyjs.com/packages/gatsby-plugin-web-font-loader/?=font) and your [Adobe Fonts project id](https://fonts.adobe.com/my_fonts#web_projects-section). For example, this is how you can add Futura to your project.
```
```suggestion
You can add Typekit Web Fonts to your project by using the [gatsby-plugin-web-font-loader](/plugins/gatsby-plugin-web-font-loader/?=font) and your [Adobe Fonts project id](https://fonts.adobe.com/my_fonts#web_projects-section). For example, this is how you can add Futura to your project.
```

Links in docs should be relative to make use of `gatsby-link`.

tyhopp reviewed (changes requested):
I did find one small thing, could you please use relative links for the case above? Thanks!
